php/proxy.php

```diff
@@ -3,150 +3,192 @@
+                $name = str_replace(' ', '-', ucwords(strtolower(str_replace('_', ' ', substr($name, 5)))));
+                $headers[$name] = $value;
+            } else if ($name == "CONTENT_TYPE") {
+                $headers["Content-Type"] = $value;
+            } else if ($name == "CONTENT_LENGTH") {
+                $headers["Content-Length"] = $value;
+            }
+        }
+        return $headers;
+    }
+
     private function getHeadersOfRequest() {
         $headers = array();
-        $isAuthToken = false; 
-        foreach (getallheaders() as $name => $value) {
-          $headerString = $name.':'.$value;
-
-          //Setting target Server as Host
-          if(strtolower($name) == 'host') {
-              $headerString = $name.':'.parse_url(self::getSettings()['url'])['host'];
-          }
-
-          if($name == 'authToken') {
-            $isAuthToken = true;
-          }
-          array_push($headers,"$headerString");
+        $isAuthToken = false;
+
+        foreach (self::emulate_getallheaders() as $name => $value) {
+
+            $headerString = $name.':'.$value;
+
+            //Setting target Server as Host
+            if(strtolower($name) == 'host') {
+                $headerString = $name.':'.$this->host;
+            }
+            if($name == 'authToken') {
+                $isAuthToken = true;
+            }
+            array_push($headers,"$headerString");
         }
-        
+
         if($isAuthToken == false){
-            $settings = self::getSettings();
-            $username = 'username:'.$settings['username'];
-            $password = 'password:'.$settings['password'];
-            array_push($headers, $username);
-            array_push($headers, $password);
-            array_push($headers,'User-Agent:Tinymce Proxy');
+            $this->settings = self::getSettings();
+            $this->username = 'username:'.$this->settings['username'];
+            $this->password = 'password:'.$this->settings['password'];
+            array_push($headers, $this->username);
+            array_push($headers, $this->password);
+            array_push($headers,'User-Agent:Drupal Proxy');
+
         }
         return $headers;
     }
-    
+
     private function getPOSTdata(){
         return file_get_contents("php://input");
     }
-    
-    private function proxyRequest($url){
+
+    public function __construct() {
+        self::disable_gzip();
+        $this->settings = self::getSettings();
+        $this->targetPath = rtrim($this->settings['url'],"/");
+        $this->host = parse_url($this->targetPath)['host'];
+
+        if(isset($_SERVER['HTTPS'] ) && $_SERVER['HTTPS'] == 'on'){
+            $this->domainURL = 'https://'.$_SERVER['HTTP_HOST'];
+        } else {
+            $this->domainURL = 'http://'.$_SERVER['HTTP_HOST'];
+        }
+
+        $part = '/acrolinx/proxy.php'; //TODO: Ensure that this is the correct location of the 'proxy.php' on the server.
```

Comment:
Is that something an integrator should do? Or we? If we a ticket might be better. If the integrator, then it's OK.

Reply:
Yes this for integrator to do, where he/she wants to place the proxy code